src/hash.rs

```diff
@@ -33,9 +33,9 @@ impl HashTable {
         }
     }
 
-    pub fn get_chain(&self, index: u32) -> u32 {
-        assert!(index < self.chain_count);
-        let index = self.bucket_count + index;
+    pub fn get_chain(&self, chain_index: u32) -> u32 {
+        assert!(chain_index < self.chain_count);
+        let index = self.bucket_count + chain_index;
```

Comment:
This doesn't seem like a good chain - it just adds duplication to the name and doesn't make anything clearer